packages/webapp/src/components-v2/EnvironmentDropdown.tsx

```diff
@@ -0,0 +1,220 @@
+import { useMeta } from '@/hooks/useMeta';
+import { useToast } from '@/hooks/useToast';
+import { useStore } from '@/store';
+
+export const EnvironmentDropdown: React.FC = () => {
+    const env = useStore((state) => state.env);
+    const setEnv = useStore((state) => state.setEnv);
+    const envs = useStore((state) => state.envs);
+    const environment = useEnvironment(env);
+    const { meta } = useMeta();
+    const [dropdownMenuOpen, setDropdownMenuOpen] = useState(false);
+    const [environmentDialogOpen, setEnvironmentDialogOpen] = useState(false);
+
+    const navigate = useNavigate();
+
+    const isMaxEnvironmentsReached = envs && environment.plan && envs.length >= environment.plan.environments_max;
+
+    const onSelect = (selected: string) => {
+        if (selected === env) {
+            return;
+        }
+
+        setEnv(selected);
+
+        const pathSegments = window.location.pathname.split('/').filter(Boolean);
+
+        pathSegments[0] = selected;
+
+        let newPath = `/${pathSegments.join('/')}`;
+
+        // If on 'integration' or 'connections' subpages beyond the second level, redirect to their parent page
+        if (pathSegments[1] === 'integrations' && pathSegments.length > 2) {
+            newPath = `/${selected}/integrations`;
+        } else if (pathSegments[1] === 'connections' && pathSegments.length > 2) {
+            newPath = `/${selected}/connections`;
+        }
+
+        navigate(newPath);
+    };
+
+    if (!meta) {
+        return;
+    }
+
+    return (
+        <SidebarMenu>
+            <SidebarMenuItem>
+                <DropdownMenu open={dropdownMenuOpen} onOpenChange={setDropdownMenuOpen} modal={false}>
+                    <DropdownMenuTrigger className="h-fit w-full rounded p-2.5 flex flex-row items-center justify-between cursor-pointer bg-dropdown-bg-default border-[0.5px] border-border-muted hover:bg-dropdown-bg-press hover:border-0 hover:border-l-[0.5px] hover:border-r-[0.5px] hover:border-r-transparent data-[state=closed]:hover:my-[0.5px] data-[state=open]:bg-dropdown-bg-press data-[state=open]:border-[0.5px] data-[state=open]:border-border-muted">
+                        <div className="flex gap-2 items-center">
+                            <LogoInverted className="h-6 w-6 text-text-primary" />
+                            <div className="flex flex-col items-start">
+                                <span className="text-s leading-3 text-text-secondary">Environment</span>
+                                <span className="text-sm leading-4 text-text-primary font-semibold capitalize truncate max-w-28">{env}</span>
+                            </div>
+                        </div>
+                        <ChevronsUpDown className="w-4.5 h-4.5 text-text-primary" />
+                    </DropdownMenuTrigger>
+                    <DropdownMenuContent align="start" side="bottom" className="w-50 flex flex-col gap-2">
```

Comment:
opinion: it's on the Figma, but I find it weird to put a spacing between buttons; it makes the hover flickers